src/components/Card.astro

```diff
@@ -20,40 +20,138 @@ const {
 } = Astro.props
 ---
 
-<article
-	class="duration-50 relative z-0 flex h-full flex-col gap-4 overflow-hidden rounded-lg border border-neutral-200 bg-white p-4 text-astro-gray-700 shadow shadow-neutral-200 transition-colors hover:border-primary-300 hover:shadow-accent-200"
->
-	<header class="flex flex-row flex-wrap justify-between gap-2">
-		<h3 class="text-lg font-bold">{title}</h3>
-		<a
-			class="ease scale-90 opacity-60 transition-all duration-100 hover:scale-100 hover:opacity-100"
-			href={sourceUrl}
-			aria-label="View Source Code"
+<article class="relative border border-astro-gray-500 bg-astro-gray-600">
+	<div class="noise z-0"></div>
+	<div class="aspect-video bg-astro-gray-700">
+		{
+			!!previewImage && (
+				<img src={previewImage} alt="" class="object-cover s-full" />
+			)
+		}
+	</div>
+	<div class="flex flex-col gap-4 p-5">
+		<h3 class="heading-4">{title}</h3>
+		<p>
+			<a href={sourceUrl} class="flex items-center gap-1">
+				<Icon name="github" size={16} />
+				<span
+					class="text-sm text-astro-gray-200 underline underline-offset-[3px]"
+					>View repo on GitHub</span
+				>
+			</a>
+		</p>
+		<hr class="border-astro-gray-500" />
+		<div
+			class="flex h-10 w-max min-w-0 max-w-full gap-px divide-x divide-astro-gray-600 rounded-full bg-blue-purple-gradient"
 		>
-			<Icon name="github" size={24} aria-hidden="true" />
-		</a>
-	</header>
-	<div class="flex flex-row flex-wrap gap-2">
-		<a href={stackblitzUrl} aria-label="Open in StackBlitz">
-			<Icon name="open-in-stackblitz" height={32} aria-hidden="true" />
-		</a>
-		<a href={codesandboxUrl} aria-label="Open in CodeSandbox">
-			<Icon name="open-in-codesandbox" height={32} aria-hidden="true" />
-		</a>
-		<a href={gitpodUrl} aria-label="Open in Gitpod">
-			<Icon name="open-in-gitpod" height={32} aria-hidden="true" />
-		</a>
+			<a
+				href={stackblitzUrl}
+				target="_blank"
+				rel="noopener noreferrer"
+				class="flex items-center gap-2 rounded-l-full px-4 text-sm transition hover:backdrop-brightness-75"
+			>
+				<Icon name="stackblitz" size={16} /> Open in StackBlitz
+			</a>
+			<details class="relative z-10" data-card-options>
```

Comment:
Excellent use of a details/summary, gotta love free keyboard support 🙌 

<img width="413" alt="Screenshot 2023-04-04 at 4 20 05 PM" src="https://user-images.githubusercontent.com/15836226/229925688-6b6df0a0-c544-4c4b-8f01-fcd74de56f28.png">

extra tiny nit: the details list can get cut off on the right edge of the screen, any easy fix for this or would it need a JS script watching it's edge position?

Reply:
I think if we align it so the right edge is below the button, it'll always be within the screen